src/spclient.rs

```diff
@@ -323,15 +346,23 @@ impl SpClient {
                 subtype: PSBT_SP_SUBTYPE,
                 key: PSBT_SP_ADDRESS_KEY.as_bytes().to_vec(),
             }) {
-                // Create the right output key
                 let sp_address = SilentPaymentAddress::try_from(deserialize::<String>(value)?)?;
-                let output_key = silentpayments::sending::generate_recipient_pubkey(
-                    sp_address.into(),
-                    partial_secret,
-                )?;
-                // update the script pubkey
-                output.script_pubkey =
-                    ScriptBuf::new_p2tr_tweaked(output_key.dangerous_assume_tweaked());
+                if let Some(xonlypubkeys) = sp_address2xonlypubkeys.get_mut(&sp_address.to_string())
```

Comment:
maybe we should also add an assert at the end that all these `xonlypubkeys` are of length 0?

Reply:
You mean to make sure that we consume all the pubkeys? I don't think there's much risk, but it's probably worth a debug_assert!